### DISCUSSION TITLE
How do I get run_id of the current run

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#8424] [Q&A] [answered]
Hey guys, is there a way to get the *run id* of the current run ?
for example, after i create a new run :
```
yield RunRequest(
    run_key="somename-{}".format(time_stamps),
    run_config={"solids": {"somename_step1": {"config": {"filename": "file"}}}},
)
```
can i get the run id that was created during the run ? i want to print logs to third party platform with the runid during the run .

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1635854645314300?thread_ts=1635854645.314300&cid=C01U954MEER).*

**Comment by yuhan** [comment, marked as answer]
You can access the run within an op/solid using the optional context arg:
```python
@op
def somename_step1(context: OpExecutionContext):
    print(context.run.run_id)
```

Check out the [API doc](https://docs.dagster.io/_apidocs/execution#dagster.OpExecutionContext) for other available information you can access using the context arg.
